File: python/rna-transcription/rna_transcription.py
def to_rna(nucleotides):
    """
    Given a DNA strand, return its transcribed RNA strand
    
    :param nucleotides: string of one or more characters in the range of G,C,T,A
    :return: string of one or more characters in the range of C,G,A,U
    """
    mapping = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}
    result = ''.join([mapping.get(nuc, '') for nuc in nucleotides])
    return result if len(result) == len(nucleotides) else ''

File: typescript/rna-transcription/rna-transcription.ts
export default class Transcriptor {
    private mapping: { [s: string]: string; } = { G: 'C', C: 'G', T: 'A', A: 'U' }

    toRna(s: string): string {
        const mapKeys = Object.keys(this.mapping)
        return s.split('').map((char: string) => {
            if (mapKeys.indexOf(char) === -1) {
                throw new Error('Invalid input DNA.')
            }
            return this.mapping[char]
        }).join('')
    }
}